apps/meteor/client/views/room/Header/RoomToolbox/RoomToolboxE2EESetup.tsx

```diff
@@ -4,19 +4,23 @@ import { useTranslation } from '@rocket.chat/ui-contexts';
 import React from 'react';
 
 import { roomActionHooksForE2EESetup } from '../../../../ui';
+import { useRoom } from '../../contexts/RoomContext';
 import type { RoomToolboxActionConfig } from '../../contexts/RoomToolboxContext';
 import { useRoomToolbox } from '../../contexts/RoomToolboxContext';
+import { getRoomGroup } from '../../lib/getRoomGroup';
 
 const RoomToolboxE2EESetup = () => {
 	const t = useTranslation();
 	const toolbox = useRoomToolbox();
+	const room = useRoom();
 
 	const { tab } = toolbox;
 
 	const actions = useStableArray(
 		roomActionHooksForE2EESetup
 			.map((roomActionHook) => roomActionHook())
-			.filter((roomAction): roomAction is RoomToolboxActionConfig => !!roomAction),
+			.filter((roomAction): roomAction is RoomToolboxActionConfig => !!roomAction)
+			.filter((action) => !action.groups || action.groups.includes(getRoomGroup(room))),
 	);
```

Comment:
Any chance these filters could be done in a single `filter` call?

Reply:
Sure, I just did this way to improve readability

Reply:
Done!